pkg/clients/database/mysql.go

```diff
@@ -336,3 +342,42 @@ func IsMySQLUpToDate(p azuredbv1beta1.SQLServerParameters, in mysql.Server) bool
 	}
 	return true
 }
+
+// TODO(muvaf): FetchAsyncOperation can be used by other managed resources as well.
+
+// FetchAsyncOperation updates the given operation object with the most up-to-date
+// status retrieved from Azure API.
+func FetchAsyncOperation(ctx context.Context, client autorest.Sender, as *azuredbv1beta1.AsyncOperation) error {
```

Comment:
Could we move this to the general `azure` client pkg and maybe move `AsyncOperation` to `azure` api types pkg so that it can be embedded in other resource types?